Refs #4478 - API localization - Apipie setup moved

mbacovsky opened the pull request:
*DO NOT MERGE*
This is suggestion to move Apipie setup to plugin atributes. This will bring us better control over the Apipie configuration e.g. this change will make it possible to pre-generate apipie cache for the plugins during the package build time with ```rake plugin:apipie:cache['katello']```

ehelms commented:
@mbacovsky is there a timeline for this work?

ehelms commented on rubygem-katello.spec:
```diff
@@ -195,8 +195,18 @@ cp -a .%{gem_dir}/* \
 
 mkdir -p ./usr/share
 cp -r %{foreman_dir} ./usr/share || echo 0
+sed -i '/production:/,$d' ./usr/share/foreman/config/database.yml
```
I think you need to create database.yml first by copying https://github.com/theforeman/foreman/blob/develop/config/database.yml.example to database.yml

mbacovsky commented:
@ehelms, I've updated and rebased the PR. I was able to build it in my local el7 mock as non-root user. Koji is having difficulties chewing it for some reason (seems stuck). I'll add a link to scratch build when it finishes.

mbacovsky commented:
Successfull scratch build: http://koji.katello.org/koji/taskinfo?taskID=233749

ehelms commented:
Looks to be building happily now but there are rubocop issues. I also noticed there are changes to migrations present that look unrelated to apipie?

ehelms commented:
@mbacovsky do all plugins have to migrate the database in order to generate the apipie cache in the spec?

mbacovsky commented:
@ehelms I fixed the rubocop issues. 
re migrations: To generate apipie cache the Apipie have to load rails env, which needs the (migrated) DB. The updated migrations were failing on `sqlite` so I rephrased them, but it should be equivalent.

daviddavis commented:
ACK

ehelms commented:
ACK - merging

mbacovsky commented:
@ehelms, @daviddavis, thanks a lot!
